silk/datadog_checks/silk/check.py

```diff
@@ -0,0 +1,238 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import deepcopy
+
+from six.moves.urllib.parse import urljoin, urlparse
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+from datadog_checks.base.utils.time import get_timestamp
+
+from .events import SilkEvent
+from .metrics import BLOCKSIZE_METRICS, METRICS, READ_WRITE_METRICS
+
+EVENT_PATH = "events?timestamp__gte={}"
+
+STATE_ENDPOINT = 'system/state'
+SERVERS_ENDPOINT = 'system/servers'
+
+STATE_MAP = {'online': AgentCheck.OK, 'offline': AgentCheck.WARNING, 'degraded': AgentCheck.CRITICAL}
+
+
+class SilkCheck(AgentCheck):
+
+    __NAMESPACE__ = 'silk'
+
+    STATE_SERVICE_CHECK = "system.state"
+    SERVERS_SERVICE_CHECK = "server.state"
+    CONNECT_SERVICE_CHECK = "can_connect"
+
+    def __init__(self, name, init_config, instances):
+        super(SilkCheck, self).__init__(name, init_config, instances)
+        self.metrics_to_collect = dict(METRICS)
+
+        server = self.instance.get("host_address")
+
+        if server is None:
+            raise ConfigurationError("host_address is a required parameter.")
+
+        self.latest_event_query = int(get_timestamp())
+        self.url = "{}/api/v2/".format(server)
+
+        host = urlparse(server).netloc
+        self._tags = self.instance.get("tags", []) + ["silk_host:{}".format(host)]
+
+        if self.instance.get("enable_read_write_statistics", False):
+            self.metrics_to_collect.update(dict(READ_WRITE_METRICS))
+
+        if self.instance.get("enable_blocksize_statistics", False):
+            self.metrics_to_collect.update(dict(BLOCKSIZE_METRICS))
+
+        # System tags are collected from the /state/endpoint
+        self._system_tags = []
+
+    def check(self, _):
+        # Get system state and tags
+        self.submit_system_state()
+        # Submit service checks for each server
+        self.submit_server_state()
+
+        metric_tags = self._tags + self._system_tags
```

Comment:
Let's pass a tags list around rather than modify an attribute at every run

Reply:
`self. _system_tags` gets filled with the system tags based on `self.submit_system_state()`, otherwise `self. _system_tags` is `[]`. Is there another way I can set `metric_tags` without modifying it every run?

Reply:
See https://github.com/DataDog/integrations-core/blob/5aa0c250a2c3bf92484034416988254f744505ff/vault/datadog_checks/vault/check.py#L55-L58